include/deal.II/fe/fe_enriched.h

```diff
@@ -644,6 +648,160 @@ class FE_Enriched : public FiniteElement<dim, spacedim>
                           internal::FEValuesImplementation::FiniteElementRelatedData<dim,spacedim>    &output_data) const;
 };
 
+
+
+namespace ColorEnriched
+{
+  /**
+   * An alias template for predicate function which returns a
+   * boolean for a Triangulation<dim>::cell_iterator object.
+   *
+   * This is used by helper functions and in the implementation of
+   * ColorEnriched::Helper class.
+   */
+  template <int dim>
+  using predicate_function
+    = std::function< bool
+      (const typename Triangulation<dim>::cell_iterator &) >;
+
+  /**
+   * ColorEnriched::Helper class provided to create hp::FECollection, a
+   * collection of finite element objects used by hp::DoFHandler in a domain
+   * with multiple, possibly overlapping, sub-domains with individual
+   * enrichment functions. Note that the overlapping regions may have
+   * multiple enrichment functions associated with it. This is implemented
+   * using a general constructor of FE_Enriched object which allows different
+   * enrichment functions.
+   *
+   * <code> Helper<dim> </code> is first constructed by providing a base
+   * FiniteElement @p fe_base, the enrichment FiniteElement @p fe_enriched,
+   * a vector of predicate functions and enrichment functions
```

Comment:
...predicate functions that define subdomains as well as the corresponding enrichment functions.